### PR TITLE
renamed limit to head. implement Fugue SQL version

### PR DESCRIPTION
**Pull request by kvnkho**
This is 90% done. Limit has now been renamed to head.

The SQL usage for head looks like:
```
a=create [[NULL, 1], [NULL, 2], [1, NULL], [1, 2]] schema a:double, b:double
head prepartition by a presort b desc nulls first rows 1 from a
```

There are just two things that I can't quite resolve myself. First is under workflow.py. WorkflowDataFrame inherits from DataFrame. DataFrame already has a `head` method. Having `head` WorkflowDataFrame causes issues with the two methods having incompatible signatures. There are two solutions here:

1. The easier one, is maybe we rename everything `Take`, like the Spark method.
2. Refactor out the `head` method from DataFrame and keep it in WorkflowDataFrame (not 100% sure of side effects)

Second, I am half-satisfied with the SQL in the .g4 file for `HEAD`. It works for now, but NULL FIRST or NULL LAST is required. I want to make it optional.
```
fugueHeadTask:
    HEAD (partition=fuguePrepartition)? (PRESORT presort=fugueColsSort)? (NULL|NULLS) na_position=(FIRST|LAST) (ROWS rows=INTEGER_VALUE)? (FROM df=fugueDataFrame)?
    ;
```

Ideally, I want `(NULL FIRST|NULL LAST)`?. I can then set the default to be NULL LAST on the Python _visitor side. The issue here is I don't think you can grab it with `ctx.NULL LAST` since there is a space in the _visitor side.

I also tried `na_position=(FIRST|LAST)?` but this breaks so many things. I have not figured out the right syntax, but this setup works.

### CODE REVIEW

**Comment by goodwanghan** [comment]
`(partition=fuguePrepartition)? (PRESORT presort=fugueColsSort)?`
I think this should be `((partition=fuguePrepartition)|(PRESORT presort=fugueColsSort))?`
Because these 2 should not exist together

`(NULL|NULLS) na_position=(FIRST|LAST)` should be optional

df is better to be after `HEAD`, this order should be similar to transform
`HEAD df PREPARTITION BY a PRESORT b DESC ROWS 5`

**Comment by kvnkho** [comment]
This is now complete. 100% coverage. The method has been renamed to take, based off of Spark.

SQL Implementation has been changed. Sample below:
`take 1 row from a prepartition by a presort b desc nulls first`

File: fugue_sql/_antlr/fugue_sql.g4 [hunk 1]
```diff
@@ -163,6 +164,10 @@ fugueSampleTask:
     SAMPLE REPLACE? method=fugueSampleMethod (SEED seed=INTEGER_VALUE)? (FROM df=fugueDataFrame)?
     ;
 
+fugueTakeTask:
+    TAKE (rows=INTEGER_VALUE (ROW|ROWS))? (FROM df=fugueDataFrame)? ((partition=fuguePrepartition)|(PRESORT presort=fugueColsSort))? ((NULL|NULLS) na_position=(FIRST|LAST))?
```
Comment (goodwanghan) [inline fugue_sql/_antlr/fugue_sql.g4:168]: nice!

**Comment by goodwanghan** [comment]
👍

**Review by goodwanghan — APPROVED** [review]
Love it!

**Pull request merged**